### PR TITLE
Migrated Tests Files

### PR DESCRIPTION
**Pull request by masmei**
#359 

-Added dependency for typescript jest. Updated config. 
-Migrated ./src/setupTests.js
-Migrated ./src/utils/test-utils.js
-Created and migrated renderWithProviderAndRouter component and updates imports in all connected files due to error inside of the .ts file.

### CODE REVIEW

**Comment by arielge** [comment]
hi @masmei,
Thank you for your contribution, and welcome to the LS community!

I ran our frontend test workflow and it appears these changes result in the following error:
```
TS7006: Parameter 'actionPromiseResult' implicitly has an 'any' type.
    68 |     const prevValue = value;
    69 |     setValue("");
  > 70 |     dispatch(deleteWorkspace({ workspaceId: value })).then((actionPromiseResult) => {
       |                                                             ^^^^^^^^^^^^^^^^^^^
```


Note that you can try building the frontend on your machine by following the instructions [here](https://github.com/label-sleuth/label-sleuth/tree/main/frontend), and run tests using `npm run test`. @martinscooper should be able to help you out if you have any questions.

Also, can you please add a copy of the [Developer's Certificate of Origin 1.1](https://elinux.org/Developer_Certificate_Of_Origin) to the PR description (as outlined in the [Contribution Guidelines](https://github.com/label-sleuth/label-sleuth/blob/main/CONTRIBUTING.md#contributor-license-agreement))? Thanks!

File: frontend/src/store/configureStore.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ import { errorReducer } from "../error/errorSlice";
 import { errorMiddleware } from "../error/errorMiddleware";
 import workspaceReducer  from "../modules/Workplace/redux";
 
-export const setupStore = (preloadedState?: any) =>
+export const setupStore = (preloadedState?: any) : any =>
```
Comment (martinscooper) [inline frontend/src/store/configureStore.ts:24]: This change is making the check to fail. Adding `any` as the return type makes objects returned by the Redux store (state and actions) to be null and thus throwing compile errors. 

The return type is inferred an so there is no need to explicity set it here. Could you remove it?

**Review by martinscooper — CHANGES_REQUESTED** [review]
Hi @masmei, welcome to the Label Sleuth community! Changes look great. Once you make the change I requested and add a copy of the [Developer's Certificate of Origin 1.1](https://elinux.org/Developer_Certificate_Of_Origin) we can merge this PR.

**Comment by martinscooper** [comment]
Hi @masmei, have you had the chance to take a look at my requested change?

**Comment by martinscooper** [comment]
Hi @masmei , have you had the chance to look into my comments?

**Comment by martinscooper** [comment]
Closing this. I will implement the rest of the work in https://github.com/label-sleuth/label-sleuth/pull/528.

**Pull request closed without merging**